eslint/eslint-patch/src/eslint-bulk-suppressions/bulk-suppressions-patch.ts

```diff
@@ -0,0 +1,295 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import { TSESTree } from '@typescript-eslint/types';
+import { spawnSync } from 'child_process';
+import fs from 'fs';
+import path from 'path';
+import * as Guards from './ast-guards';
+
+import { eslintFolder } from '../_patch-base';
+
+interface Suppression {
+  file: string;
+  scopeId: string;
+  rule: string;
+}
+
+interface BulkSuppressionsJson {
+  suppressions: Suppression[];
+}
+
+function getNodeName(node: TSESTree.Node): string | null {
+  if (!Guards.isNodeWithName(node)) return null;
+
+  if (Guards.isClassDeclarationWithName(node)) return node.id.name;
+
+  if (Guards.isFunctionDeclarationWithName(node)) return node.id.name;
+
+  if (Guards.isClassExpressionWithName(node)) return node.id.name;
+
+  if (Guards.isFunctionExpressionWithName(node)) return node.id.name;
+
+  if (Guards.isNormalVariableDeclaratorWithAnonymousExpressionAssigned(node)) return node.id.name;
+
+  if (Guards.isNormalObjectPropertyWithAnonymousExpressionAssigned(node)) return node.key.name;
+
+  if (Guards.isNormalClassPropertyDefinitionWithAnonymousExpressionAssigned(node)) return node.key.name;
+
+  if (Guards.isNormalAssignmentPatternWithAnonymousExpressionAssigned(node)) return node.left.name;
+
+  if (Guards.isNormalMethodDefinition(node)) return node.key.name;
+
+  if (Guards.isTSEnumDeclaration(node)) return node.id.name;
+
+  if (Guards.isTSInterfaceDeclaration(node)) return node.id.name;
+
+  if (Guards.isTSTypeAliasDeclaration(node)) return node.id.name;
```

Comment:
At least one of prettier or ESLint should be complaining about the lack of braces on these if statements. This doesn't conform to the rushstack code formatting standards.

Reply:
see above

Reply:
It's not a part of Prettier, and the corresponding eslint rule is https://eslint.org/docs/latest/rules/curly which I believe is not turned on at the moment. There are remnants of tslint with the curly rule on.

If needed, I can go around adding braces manually.